core/src/mledb/mledb-match/mledb-match.service.ts

```diff
@@ -156,4 +168,171 @@ export class MledbMatchService {
+            if (isNcp && !winningTeam) {
+                this.logger.error("When NCPing a series associated with a fixture, you must specify a winningTeam");
+                return "Failure";
+            }
+        
+            // Check to make sure the winning team played in the series/fixture
+            if (winningTeam
+                    && series.fixture.homeName !== winningTeam.name
+                    && series.fixture.awayName !== winningTeam.name) {
+                this.logger.error(`The team \`${winningTeam?.name}\` did not play in series with id \`${series.id}\` (${series.fixture.awayName} v. ${series.fixture.homeName}), and therefore cannot be marked as the winner of this NCP. Cancelling process with no action taken.`);
+                return "Failure";
+            }
+        
+        } else if (seriesType === SeriesType.Scrim) {
+            if (!series.scrim) {
+                this.logger.error(`Error: series with id=\`${seriesId}\` has no associated scrim. If this is a league play series, please use \`!ncpSeries\` or \`!unNcpSeries\`. If this actually is a scrim, ping Bot Team.`);
+                return "Failure";
+            }
+
+            winningTeam = await this.teamRepo.findOneOrFail({
+                where: {
+                    name: "FA",
+                },
+            });
+        } else {
+            throw new Error();
+        }
+
+        const seriesReplays = await this.seriesReplayRepo.find({
+            where: {
+                series: {
+                    id: seriesId,
+                },
+            },
+        });
+
+        // Set series to Full NCP and set winning team
+        series.fullNcp = isNcp;
+        await this.seriesRepo.save(series);
+
+        // Update each series replay (including any dummies) to NCP
+        const replayIds = seriesReplays.map(seriesReplay => seriesReplay.id);
+        await this.markReplaysNcp(replayIds, isNcp, winningTeam ? winningTeam : undefined);
+
+        this.logger.debug(`(${r}) end markSeriesNcp`);
+
+        return `seriesId=${seriesId} successfully marked fullNcp=${isNcp} with updated elo, and all connected replays had their elo updated.`;
+    }
+
+    /**
+     * Marks replays as NCP=true/false, and updates the associated Elo of those replays and all connected replays accordingly.
+     * "Connected" replays are where replays in which one of the player's in the NCP replay has played. Since the NCP replay will have its elo affects removed,
+     * all subsequent replays where those player's played need to be recalculated.
+     * @param replayId The replay to mark NCP=true/false.
+     * @param isNcp Whether the given replayId should be marked NCP or un-NCP.
+     * @returns A string containing status of what was updated.
+     */
+    async markReplaysNcp(replayIds: number[], isNcp: boolean, winningTeam?: MLE_Team): Promise<string> {
+        const r = Math.floor(Math.random() * 10000);
```

Comment:
What's the significance of this? Why do we need to gen a random Id when it's not used for anything other than logs?

Reply:
Really no significance, just copied from MLEDB. We can safely remove.

Reply:
Removed